src/wasm-binary.h

```diff
@@ -1398,12 +1398,14 @@ class WasmBinaryBuilder {
                           Address defaultIfNoMax);
   void readImports();
 
-  // The signatures of each function, given in the function section
-  std::vector<Signature> functionSignatures;
+  // The signatures of each function, including imported functions, given in the
+  // import and function sections. Store HeapTypes instead of Signatures because
+  // reconstructing the HeapTypes from the Signatures is expensive.
+  std::vector<HeapType> functionTypes;
 
   void readFunctionSignatures();
-  Signature getSignatureByFunctionIndex(Index index);
-  Signature getSignatureByTypeIndex(Index index);
+  HeapType getTypeByFunctionIndex(Index index);
+  HeapType getSigTypeByIndex(Index index);
```

Comment:
I made this `getSigTypeByIndex` instead of just `getTypeByIndex` because it still checks whether the type is a signature type and fails if it is not.

Reply:
How about keeping `getSignature*` then, as convenience methods, that do the `.getSignature()` for us? Shorter to write, and clearer, I think.

Reply:
Sure, that makes sense.